src/hooks/carioHooks.ts

```diff
@@ -0,0 +1,146 @@
+import { NFTData, PutData, PutStatus } from "components/wallet/NFTData";
+import { getStarknet } from "get-starknet";
+import { uint256 } from "starknet";
+import { callContract, createContract } from "utils/blockchain/starknet"
+import optionsCompiledContract from "../compiledcairo/erc721_option.json";
+import { getNfts } from "./apiHooks";
+import { BN } from 'bn.js'
+
+const optionsContractAddress = '0x02e6a26d2fcb7256934c822ad8a81ee40aed922b271495d8eb1e05d031192f52';
+
+
+let bidsCache: PutData[] = [];
+
+export async function getAllBids() : Promise<PutData[]> {
+  // would be good to generalise these to a deployement file
+  const optioncontract = createContract(optionsContractAddress, optionsCompiledContract.abi);
+  const view_bids_count = await callContract(optioncontract, 'view_bids_count');
+
+  let all_bids: PutData[] = [];
+
+  for (let i = 0; i < view_bids_count[0]; i++) {
+    const bid_result = await callContract(optioncontract, 'view_bid', i.toString())
+    const mapped_data = bid_result.map((option: any) => {
+        const data: PutData = {
+            strike_price: uint256.uint256ToBN(option.params.strike_price).toString(10),
+            expiry_date: option.params.expiry_date.toString(10),
+            erc721_address: option.params.erc721_address.toString(16),
+            erc721_id: uint256.uint256ToBN(option.params.erc721_id).toString(10),
+            premium: uint256.uint256ToBN(option.params.premium).toString(10),
+            buyer_address: option.buyer_address.toString(16),
+            seller_address: option.seller_address.toString(16),
+            status: option.status.toNumber(),
+            bid_id: option.bid_id.toString(10)
+        };
+        return data
+    });
+    all_bids.push(...mapped_data)
+  }
+
+  bidsCache.push(...all_bids);
+  console.log('all data  ---> ' + JSON.stringify(all_bids));
+  return all_bids;
+}
+
+
+export async function useMyNFTS(user: string) : Promise<NFTData[]> {
+  const nfts = await getNfts(user);
+
+  let localCache = bidsCache;
+
+  if (localCache.length === 0) {
+    localCache = await getAllBids();
+    bidsCache.push(...await getAllBids()); // update cache
```

Comment:
Can this just be `bidsCache.push(...localCache)` instead of a call again?

Reply:
yep good spot